tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -434,6 +434,150 @@ object SQLPlanParser extends Logging {
     parsedExpressions.distinct.toArray
   }
 
+   // This parser is used for BroadcastHashJoin, ShuffledHashJoin and SortMergeJoin
+   def parseEquijoinsExpressions(exprStr: String): (Array[String], Boolean) = {
+     // ShuffledHashJoin [name#11, CEIL(DEPT#12)], [name#28, CEIL(DEPT_ID#27)], Inner, BuildLeft
+     // SortMergeJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner
+     // BroadcastHashJoin [name#11, CEIL(dept#12)], [name#28, CEIL(dept_id#27)], Inner,
+     // BuildRight, false
+     val parsedExpressions = ArrayBuffer[String]()
+     val pattern = """\[([\w#, +*\\\-\.<>=\`\(\)]+\])""".r
+     // Get all the join expressions and split it with delimiter :: so that it could be used to
+     // parse function names (if present) later.
+     val joinExprs = pattern.findAllMatchIn(exprStr).mkString("::")
+     // Get joinType and buildSide(if applicable)
+     val joinParams = pattern.replaceAllIn(exprStr, "").split(",").map(_.trim).filter(_.nonEmpty)
+     val joinType = joinParams(0).trim
+     // SortMergeJoin doesn't have buildSide, assign empty string in that case
+     val buildSide = if (joinParams.size > 1) {
+       joinParams(1).trim
+     } else {
+       ""
+     }
+     // Get individual expressions which is later used to get the function names.
+     val expressions = joinExprs.split("::").map(_.trim).map(
+       _.replaceAll("""^\[+""", "").replaceAll("""\]+$""", "")).
```

Comment:
if this is called a lot, create regex patterns. 

Probably can do it in a single replaceAll with an OR-ed regex?
```suggestion
       _.replaceAll("""^\[+|\]+$""", "")
```